### PR TITLE
change to factory syntax

### LINKED ISSUE #928: Update Initializers to class factory syntax lll
## First Time?

This is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**.

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, consider leaving this one for someone new, and looking through our general [help wanted](https://github.com/publiclab/mapknitter/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

## The problem

**What's supposed to happen:**

From the official [Leaflet docs](https://leafletjs.com/reference-1.0.0.html):
<blockquote>

Class Factories
You may have noticed that Leaflet objects are created without using the new keyword. This is achieved by complementing each class with a lowercase factory method:

new L.Map('map'); // becomes:
L.map('map');

</blockquote>

**What actually does happen:**

We use Leaflet's built-in `L.FeatureGroup` and `L.GeoJSON`, but we don't take advantage of their suggested class factory syntax. 

## Solution

1. remove the `new` keyword and change the initializer to `L.featureGroup`
https://github.com/publiclab/mapknitter/blob/204dd5e079e830342510c6ad9b84f78d0a0c55df/app/assets/javascripts/mapknitter/Annotations.js#L10

2. remove the `new` keyword and change the initializer to `L.geoJSON`
https://github.com/publiclab/mapknitter/blob/204dd5e079e830342510c6ad9b84f78d0a0c55df/app/assets/javascripts/mapknitter/Annotations.js#L29-L40

Thanks!!

## Step by Step
- [ ] Claim this issue with a comment here, below, and ask any clarifying questions you need
- [ ] Fork the repository and set it up locally following the main repo [README instructions](https://github.com/publiclab/mapknitter/blob/main/README.md)
 - [ ] Create a new feature branch with a unique name descriptive to the issue
- [ ] Try to fix the issue following the steps above, but even before you're done, you can:
  commit your changes to your branch and start a pull request (see [contributing to Public Lab software](https://publiclab.org/wiki/contributing-to-public-lab-software)) but mark it as "in progress" if you have questions or if you haven't finished
- [ ] Reference this issue in your pull request body
- [ ] Once you submit your pull request, an additional checklist will be provided for getting it merged

### Please keep us updated
💬⏰ - We encourage contributors to be respectful to the community and provide an update **within a week** of claiming a first-timers-only issue. We're happy to keep it assigned to you as long as you need if you update us with a request for more time or help, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance! 

If this happens to you, don't sweat it! Grab another open issue.

##  💬   Get help
If you need any help - here are some options: 

- Comment below
- Join our gitter chat at https://gitter.im/publiclab/publiclab

### PR DESCRIPTION
**Pull request by DraconianLore**
Fixes #928 (<=== Add issue number here)
https://github.com/publiclab/mapknitter/issues/928

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

[//]: # (To mark checkbox write 'x' within the square brackets)

* [x ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ x] screenshots/GIFs are attached 📎 in case of UI updation
* [ x] ask `@publiclab/mapknitter-reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by DraconianLore** [comment]
> @DraconianLore This is great, but you have made some extra edits beyond what is requested in original PR description. I would suggest you just reset your branch back to the beginning and make the changes specific to this PR again.
> 
> Let me know if you need any help with this!

Hi @sashadev-sky I seem to still be having problems now that I reset back to the start and only pushed the one file I was supposed to be changing, I cant quite figure out what I am doing wrong.

File: app/assets/javascripts/mapknitter/Annotations.js [hunk 1]
```diff
@@ -4,10 +4,10 @@ MapKnitter.Annotations.include({
     MapKnitter.Resources.prototype.initialize.call(this, options);
 
     var map = options.map,
-        drawOptions = {};
+      drawOptions = {};
 
     this._map = map;
-    this._drawnItems = new L.FeatureGroup().addTo(map);
+    this._drawnItems = = L.featureGroup().addTo(map);
```
Comment (sashadev-sky) [inline app/assets/javascripts/mapknitter/Annotations.js]: @DraconianLore you're rebase was perfect! You need to fix an issue here now, you accidentally included 2 spaces. You can commit this suggestion directly if you prefer.

```suggestion
    this._drawnItems = L.featureGroup().addTo(map);
```

**Comment by sashadev-sky** [comment]
@DraconianLore ok so that fixed one of the errors! One left -- will look into this

**Comment by sashadev-sky** [comment]
@DraconianLore locally works fine for me with your changes - what is your result when you run `$ rails test:controllers` locally ?

**Comment by DraconianLore** [comment]
@sashadev-sky its all passing on mine
`91 tests, 274 assertions, 0 failures, 0 errors, 1 skips`

File: app/assets/javascripts/mapknitter/Annotations.js [hunk 2]
```diff
@@ -4,10 +4,10 @@ MapKnitter.Annotations.include({
     MapKnitter.Resources.prototype.initialize.call(this, options);
 
     var map = options.map,
-        drawOptions = {};
+      drawOptions = {};
```
Comment (gauravano) [inline app/assets/javascripts/mapknitter/Annotations.js]:
```suggestion
    drawOptions = {};
```

Comment (gauravano) [reply app/assets/javascripts/mapknitter/Annotations.js]: @DraconianLore just a small change and we're good to go. Thanks!!

**Comment by gauravano** [comment]
Re-triggered build and it passed ✅

**Pull request merged**